src/middlewares/userMiddleware.js

```
@@ -0,0 +1,17 @@
+import userValidation from '../validation/index';
+
+/**
+ * Middleware for input validations
+ */
+export default class ValidationMiddleware {
+/**
+     * Middleware for user validation during signup/registration
+     * @param {object} req - The request from the endpoint.
+     * @param {object} res - The response returned by the method.
+     * @param {object} next - the returned values going into the next operation.
+     * @returns {object} - returns an object (error or response).
+     */
+  static onSignup(req, res, next) {
+    userValidation.signup(req, res, next);
```

Uservalidation should either throw an error in the form of ApiError when there's an error in validation or return true when all validation passes.
Sending back responses to the user should be done in the middleware.
The method in the middleware could be renamed to be more descriptive e.g `static validateSignup()`

---

The error responses are already handled in the userValidation call. I don't see why this has to be done specifically in the middleware class. All error handling occurs in the validation class already (on success and on failure).